src/main/java/com/github/dockerjava/core/util/CompressArchiveUtil.java

```diff
@@ -28,14 +28,19 @@ private CompressArchiveUtil() {
         // utility class
     }
 
-    static void putTarEntry(TarArchiveOutputStream tarOutputStream, TarArchiveEntry tarEntry, Path file)
+    static void addFileToTar(TarArchiveOutputStream tarArchiveOutputStream, Path file, String entryName)
             throws IOException {
-        tarEntry.setSize(Files.size(file));
```

Comment:
In case you wonder, no need to set the size by hand when using TarArchiveOutputStream.createArchiveEntry method with takes care of setting the file size of the entry and more, like file modification time.

Reply:
Could you PR your changes in one PR? Difficult to track in mind 5-6 PRs

Reply:
Sure, I thought it would be easier to review, let me close this one and open another one